src/transformStateWithClones.js

```diff
@@ -7,7 +7,55 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const finalState = [];
+  let newState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const action1 = actions[i];
+
+    switch (action1.type) {
+      case 'addProperties':
+        const addedResults = addProperties(action1, newState);
+
+        finalState.push(addedResults);
```

Comment:
Avoid intermediate variables, where you don't need them

```suggestion
        finalState.push( addProperties(action1, newState));
```